File: menuapp/lib/Model/hoaDon.dart
import 'package:menuapp/Model/chitiethoadon.dart';

class HoaDon {
  int hoaDonId;
  int khachHangId;
  int banId;
  DateTime ngayTao;
  int tongTien;
  List<ChiTietHoaDonDTO> chiTietHoaDons;

  HoaDon({
    required this.hoaDonId,
    required this.khachHangId,
    required this.banId,
    required this.ngayTao,
    required this.tongTien,
    required this.chiTietHoaDons,
  });

  factory HoaDon.fromJson(Map<String, dynamic> json) {
    return HoaDon(
      hoaDonId: json['hoaDonId'],
      khachHangId: json['khachHangId'],
      banId: json['banId'],
      ngayTao: DateTime.parse(json['ngayTao']),
      tongTien: json['tongTien'].toDouble(),
      chiTietHoaDons: List<ChiTietHoaDonDTO>.from(
          json['chiTietHoaDons'].map((x) => ChiTietHoaDonDTO.fromJson(x))),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'hoaDonId': hoaDonId,
      'khachHangId': khachHangId,
      'banId': banId,
      'ngayTao': ngayTao.toIso8601String(),
      'tongTien': tongTien,
      'chiTietHoaDons': List<dynamic>.from(chiTietHoaDons.map((x) => x.toJson())),
    };
  }
}

File: ADMIN/ADMIN/Repository/Models/HoaDon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace ADMIN.Repository.Models
{
    [Table("hoa_don")]
    public partial class HoaDon
    {
        public HoaDon()
        {
            ChiTietHoaDons = new HashSet<ChiTietHoaDon>();
        }

        [Key]
        [Column("hoa_don_id")]
        public int HoaDonId { get; set; }
        [Column("khach_hang_id")]
        public int KhachHangId { get; set; }
        [Column("ban_id")]
        public int BanId { get; set; }
        [Column("ngay_tao", TypeName = "datetime")]
        public DateTime NgayTao { get; set; }
        [Column("tong_tien")]
        public int TongTien { get; set; }

        [ForeignKey("BanId")]
        [InverseProperty("HoaDons")]
        public virtual Ban Ban { get; set; } = null!;
        [ForeignKey("KhachHangId")]
        [InverseProperty("HoaDons")]
        public virtual KhachHang KhachHang { get; set; } = null!;
        [InverseProperty("HoaDon")]
        public virtual ICollection<ChiTietHoaDon> ChiTietHoaDons { get; set; }
    }
}
